client/layout/guided-tours/tours/simple-payments-email-tour.js

```diff
@@ -33,6 +34,46 @@ const handleTargetDisappear = () => {
 	tourFirstStep.style.left = '-9999px';
 };
 
+// IE9+ polyfill for `Element.matches()` used in `DelegatingQuit`
+if ( ! Element.prototype.matches ) {
+	Element.prototype.matches = Element.prototype.msMatchesSelector;
+}
+
+class DelegatingQuit extends Quit {
+	addTargetListener = () => {
+		const { parentTarget } = this.props;
+		const container = targetForSlug( parentTarget );
+
+		if ( container && container.addEventListener ) {
+			container.addEventListener( 'click', this.onClick );
+			container.addEventListener( 'touchstart', this.onClick );
+		}
+	};
+
+	removeTargetListener = () => {
+		const { parentTarget } = this.props;
+		const container = targetForSlug( parentTarget );
+
+		if ( container && container.addEventListener ) {
+			container.removeEventListener( 'click', this.onClick );
+			container.removeEventListener( 'touchstart', this.onClick );
+		}
+	};
+
+	onClick = event => {
+		let eventTarget = event.target;
+		while ( eventTarget !== event.currentTarget ) {
+			if ( eventTarget.matches( this.props.target ) ) {
+				this.props.onClick && this.props.onClick( event );
+				const { quit, tour, tourVersion, step, isLastStep } = this.context;
+				quit( { tour, tourVersion, step, isLastStep } );
+				return;
+			}
+			eventTarget = eventTarget.parentNode;
```

Comment:
Could `eventTarget.parentNode` ever be `null` and should we guard against that case?

Reply:
It could happen if and only if `props.target` is not a descendant of `props.parentTarget`.
Basically, so, only if either it was improperly set or if the DOM changed somehow.
Yours is a good call especially for the second case, and thinking of how the whole editor could change sooner or later with Gutenberg.
I'll add a `return` if `eventTarget` is falsey.